tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,496 @@
+        # Checking whether implementation raising the TypeError Exception for invalid comms object
+        with self.assertRaises(TypeError):
+            AWSIoT("Invalid object", enclose_metadata=True)
+
+    def test_implementation_register_entity(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Register the edge
+        registered_entity = self.aws.register(self.edge_system)
+
+        # Check the returned object is of the class RegisteredEntity
+        self.assertIsInstance(registered_entity, RegisteredEntity)
+
+    def test_implementation_register_metric(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = self.aws.register(test_metric)
+
+        # Check the returned object is of the class RegisteredMetric
+        self.assertIsInstance(registered_metric, RegisteredMetric)
+
+    def test_implementation_create_relationship(self):
+        """
+        Method to test RegisteredEntity and Parent and RegisteredMetric as child.
+        :return: None
+        """
+
+        # Register the edge
+        registered_entity = self.aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = self.aws.register(test_metric)
+
+        # Creating the parent-child relationship
+        self.aws.create_relationship(registered_entity, registered_metric)
+
+        self.assertEqual(registered_metric.parent, registered_entity, "Check the implementation of create_relationship")
+
+    def test_validation_create_relationship_metric_device(self):
+        """
+        Method to test RegisteredMetric as Parent and RegisteredEntity as child.
+        :return: None
+        """
+
+        # Register the edge
+        registered_entity = self.aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = self.aws.register(test_metric)
+
+        with self.assertRaises(TypeError):
+            # Test case to check RegisteredEntity as both Parent and Child
+            self.aws.create_relationship(registered_metric, registered_entity)
+
+    def test_validation_create_relationship_child_entity(self):
+        """
```

Comment:
Test case to check RegisteredEntity as Parent and Child.

The reason to provide comments is that `it should describe what this test case is actually checking`.  From the test case name itself it is evident that we are testing `create_ralationship()` method.  There is no use in providing the same in comments.

Please ensure this in all of the test cases.  It is very difficult to go-through each and every test case and understand what it is doing and then review it.